common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -70,21 +70,9 @@
 <%def name="require_module(module_name, class_name)">
     <script type="text/javascript">
         (function (require) {
-            % if settings.REQUIRE_DEBUG:
-                require(['${module_name}'], function (${class_name}) {
-                    ${caller.body()}
-                });
-            % else:
-                ## The "raw" parameter is specified to avoid the URL from being further maninpulated by
-                ## static_replace calls (as woudl happen if require_module is used within courseware).
-                ## Without specifying "raw", a call to static_replace would result in the MD5 hash being
-                ## being appended more than once, causing the import to fail in production environments.
-                require(['${staticfiles_storage.url(module_name + ".js") + "?raw"}'], function () {
-                    require(['${module_name}'], function (${class_name}) {
-                        ${caller.body()}
-                    });
-                });
-            % endif
+            require(['${module_name}'], function (${class_name}) {
```

Comment:
This is an interesting change. It greatly simplifies this logic, but at the expense of having to register every factory file in ``REQUIRE_JS_PATH_OVERRIDES``. There's quite a good chance that as folks add new factories, they won't remember to add them to the list, and then we'll end up with uncached files on prod. If we do want to go down this path, I recommend adding a comment where the factories are defined here:

https://github.com/edx/edx-platform/blob/master/lms/static/lms/js/build.js#L20

For the sake of a DRY implementation, I wonder if it is possible to have a single list of all the factories, and have that feed both the RequireJS bundle configuration and the ``REQUIRE_JS_PATH_OVERRIDES``. What do you think?

I was worried that you hadn't made a corresponding change to Studio, but then when I investigated I realized that it doesn't matter because we still put a Git hash in the static URLs. We do plan to switch this over to the LMS scheme some day, at which point we'll need to have a similar list of overrides that we manage there (https://openedx.atlassian.net/browse/TNL-2680).

It is great that this solves the issue with needing the ugly ``raw`` parameter. I would like to have @cahrens verify that her use case still works with this change.